docs/source/references/modules.rst

```diff
@@ -12,6 +12,47 @@ metrics.
 .. autoclass:: torchmetrics.Metric
     :noindex:
 
+.. note::
+
+    Metric states are being synchornized by default when ``persistent=True`` in a distributed setting
+    when using :meth:`~torchmetrics.Metric.state_dict`
+    This behaviour can be blocked in the following ways.
+    We strongly recommend calling :meth:`~torchmetrics.Metric.state_dict` on all processes or this can cause a deadlock.
```

Comment:
I don't think this should be a note, it's an important detail


```suggestion
Metric states are synchornized when ``persistent=True`` in a distributed setting
when calling :meth:`~torchmetrics.Metric.state_dict`
We strongly recommend calling :meth:`~torchmetrics.Metric.state_dict` on all processes, otherwise this can cause a deadlock.
This can also happen when the ``state_dict`` of your :class:`torch.nn.Module` is generated if metric attributes have been registered.
```